pkg/dao/external_resource.go

```diff
@@ -1,41 +1,70 @@
 package dao
 
 import (
+	"fmt"
+	"io/ioutil"
 	"net/http"
 	"net/url"
 	"path"
 	"time"
 )
 
+const RequestTimeout = 3 * time.Second
+
 type ExternalResourceDaoImpl struct {
 }
 
 func GetExternalResourceDao() ExternalResourceDao {
 	return ExternalResourceDaoImpl{}
 }
 
-// ValidRepoMD Does a HEAD request on url/repodata/repomd.xml
-//  and returns any error and HTTP code encountered
-//  Uses a very short timeout, as this is intended for a
-//  small test of validity.  Actual fetching will use a longer timeout
-func (erd ExternalResourceDaoImpl) ValidRepoMD(url string) (int, error) {
+// fetchFile helper function for fetching metadata files within a repository
+func (erd ExternalResourceDaoImpl) fetchFile(url string) (*string, int, error) {
+	var sigBody string
 	var code int
-	timeout := 3 * time.Second
-	transport := http.Transport{ResponseHeaderTimeout: timeout}
-	client := http.Client{Transport: &transport, Timeout: timeout}
 
-	url, err := UrlToRepomdUrl(url)
-	if err != nil {
-		return 0, err
-	}
-	resp, err := client.Head(url)
+	transport := http.Transport{ResponseHeaderTimeout: RequestTimeout}
+	client := http.Client{Transport: &transport, Timeout: RequestTimeout}
+
+	resp, err := client.Get(url)
 
 	if err == nil {
+		bytes, err := ioutil.ReadAll(resp.Body)
+		sigBody = string(bytes)
 		code = resp.StatusCode
 		resp.Body.Close()
+		return &sigBody, code, err
+	} else {
+		return nil, code, err
 	}
+}
 
-	return code, err
+// FetchSignature fetches the yum metadata signature
+//  and returns any error and HTTP code encountered along with the contents.
+//  Uses a very short timeout, as this is intended for a
+//  small test of validity.  Actual fetching will use a longer timeout
```

Comment:
Here it says, "actual fetching will use a longer timeout". When does actual fetching occur?

Reply:
during introspection